dependent/stats_containers.h

```diff
@@ -0,0 +1,56 @@
+#ifndef _DEPENDENT_STATS_CONTAINERS_H_
+#define _DEPENDENT_STATS_CONTAINERS_H_
+
+#include <functional>
+#include <string>
+#include <string_view>
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+
+#include "dependent/stats_allocator.h"
+
+namespace std {
+
+// Dealing with libstdc++ lack of support for hashing strings with a non-default
+// allocator.
+template <typename Tag, typename Char, typename CharTraits>
+class hash<std::basic_string<Char, CharTraits,
+                             dependent::stats_allocator<Char, Tag>>> {
+  using sv = std::basic_string_view<Char, CharTraits>;
+
+ public:
+  constexpr std::size_t operator()(sv s) const { return std::hash<sv>{}(s); }
+};
+
+}  // namespace std
+
+namespace dependent {
+
+template <typename Tag>
+struct stats_containers {
```

Comment:
All contents of this file seems to be useful only for testing needs. So lets move it to test_utils folder.

Reply:
I think it might be helpful for instrumenting something used in production.
Even though, I agree that this is not an ideal place for it.
Any alternatives?

Reply:
We can create something like dependent/utils/ for that.

Reply:
Yeah, OK.